test_conformance/compiler/test_build_helpers.cpp

```diff
@@ -1009,3 +1017,127 @@ REGISTER_TEST(get_program_build_info)
 
     return 0;
 }
+
+cl_int test_kernel_name_len(cl_context context, cl_device_id device,
+                            const cl_uint length)
+{
+    cl_int error = CL_SUCCESS;
+
+    std::string buf = { "abcdefghijklmnopqrstuvwxyz" };
+    std::string name;
+    name.reserve(length);
+
+    for (cl_uint i = 0; i < length; ++i) name += buf[i % buf.size()];
+
+    std::string program_source =
+        str_sprintf(std::string(sample_name_size_test_kernel), name.c_str());
+    const char *ptr = program_source.c_str();
+
+    {
+        clProgramWrapper program =
+            clCreateProgramWithSource(context, 1, &ptr, nullptr, &error);
+        if (error != CL_SUCCESS)
+        {
+            log_error("ERROR: Unable to create program with length of "
+                      "kernel name "
+                      "%d : %s! (%s from %s:%d)\n",
+                      length, name.c_str(), IGetErrorString(error), __FILE__,
+                      __LINE__);
+            return TEST_FAIL;
+        }
+
+        error = clBuildProgram(program, 1, &device, nullptr, nullptr, nullptr);
+        if (error != CL_SUCCESS)
+        {
+            log_error("ERROR: Unable to build test program with length of "
+                      "kernel name "
+                      "%d :\n%s! (%s from %s:%d)\n",
+                      length, ptr, IGetErrorString(error), __FILE__, __LINE__);
+            return TEST_FAIL;
+        }
+
+        clKernelWrapper kernel = clCreateKernel(program, name.c_str(), &error);
+        if (error != CL_SUCCESS)
+        {
+            log_error("ERROR: Unable to create kernel with length of "
+                      "kernel name "
+                      "%d : %s! (%s from %s:%d)\n",
+                      length, name.c_str(), IGetErrorString(error), __FILE__,
+                      __LINE__);
+            return TEST_FAIL;
+        }
+
+        // query kernel name
+        size_t kernel_name_size = 0;
+        error = clGetKernelInfo(kernel, CL_KERNEL_FUNCTION_NAME, 0, nullptr,
+                                &kernel_name_size);
+        test_error(error, "clGetKernelInfo (size) failed");
+
+        std::vector<char> kernel_name(kernel_name_size);
+        error = clGetKernelInfo(kernel, CL_KERNEL_FUNCTION_NAME,
+                                kernel_name_size, kernel_name.data(), nullptr);
+        test_error(error, "clGetKernelInfo (name) failed");
+
+        if (name != std::string(kernel_name.data()))
+        {
+            log_error("Kernel name mismatch! expected=%s got=%s\n",
+                      name.c_str(), kernel_name.data());
+            return TEST_FAIL;
+        }
+    }
+
+#if defined(CL_VERSION_1_2)
+    {
```

Comment:
The CTS should always be built with the latest headers, so we shouldn't need to check for this define.  Do you intend to check whether the device supports OpenCL 1.2 here?